compiler/noirc_evaluator/src/ssa/opt/unrolling.rs

```diff
@@ -465,9 +465,14 @@ impl<'f> LoopIteration<'f> {
         // instances of the induction variable or any values that were changed as a result
         // of the new induction variable value.
         for instruction in instructions {
-            self.inserter.push_instruction(instruction, self.insert_block);
+            // Skip reference count instructions since they are only used for brillig, and brillig code is not unrolled
+            if !matches!(
```

Comment:
Can you add this to the top level module comment explaining that this pass is not usable for brillig because it removes inc/dec rcs as well?

I think it's feasible that we may one day want to unroll some loops in brillig so we may want to note this for then.